### ISSUE TITLE
[BUDI-6799] Checkbox default value not working when referencing data provider

### ISSUE THREAD

**Issue by agmcnabb** [Budibase/budibase#10119]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
- Self
  - Method: docker single image
  - Budibase Version: 2.4.31
  - App Version: 2.4.31

**Describe the bug**
In a simple nest of Data Provider > Form > Checkbox, the checkbox will not resolve to its correct default value if the JS references a field in the data provider.

**To Reproduce**
1. Create a new app.
2. Add a Data Provider, Form and Checkbox component.
3. Set the Data Provider source to any table (I have tested this with internal tables and MySQL tables)
4. In the Checkbox's default value field, enter the following JS:
`let test = $("Data Provider.Rows")[0].testField
return true` where testField is any field of the chosen table.

**Expected behavior**
In preview mode or production mode, the checkbox should be ticked on load, but it is unticked. The following are also true:
- A paragraph component with the same JS displays 'true' as expected.
- The devtools in preview mode show that the checkbox's default value is 'true' as expected.

**Screenshots**
Component nest:
![image](https://user-images.githubusercontent.com/125557661/227896494-6f185924-362b-4ae5-9e46-f2dae88199ca.png)

Table:
![image](https://user-images.githubusercontent.com/125557661/227896661-1e15103a-ae30-412a-886d-17b1b66fefb4.png)

Default value JS:
![image](https://user-images.githubusercontent.com/125557661/227896775-cea77c85-eabe-4e7f-a7e8-dd62b3fb463f.png)

Behaviour:
![image](https://user-images.githubusercontent.com/125557661/227896871-d9ee51ad-26c6-4458-85d7-9aa06430f8e2.png)

Devtools:
![image](https://user-images.githubusercontent.com/125557661/227896970-c5603632-58c3-4131-a58b-7562332a7e45.png)

  
**App Export**
[test app-export-1679908342735.tar.gz](https://github.com/Budibase/budibase/files/11076809/test.app-export-1679908342735.tar.gz)

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro
 - Browser: Chrome
 - Version: 111

**Additional context**
I had this issue a couple of months ago and I was able to work around it by using the following JS in the default value field:
`return ($("Data Provider.Rows")[0].testField.toString() == 1) ? true : false`
But this no longer works.

<sub>[BUDI-6799](https://linear.app/budibase/issue/BUDI-6799/checkbox-default-value-not-working-when-referencing-data-provider)</sub>

**Comment by agmcnabb** [comment]
Update: I'm seeing the same behaviour with option pickers. An option picker with options 'yes' and 'no', displays as 'null' if I include a reference to a data provider field in the Default Value JS. If the picker is set to radio buttons, neither button is checked. Devtools similarly confirm that the default value should be 'yes'.

![image](https://user-images.githubusercontent.com/125557661/228561018-d4e52de6-3408-461f-978f-8ed867247168.png)

**Comment by AlejandroFNadal** [comment]
I can confirm this bug, browser Firefox OS Linux.

If I replace that variable with 1, it works. 

Default value for checkbok:
```let checkData = $("Stock_Item_DP.Rows.0.Rated")
if(checkData === 1 || checkData === "1" || checkData === true || checkData === "true"){
	return true;
} else {
	return false;
}```

**Comment by aptkingston** [comment]
I am unable to reproduce this, even using the app export provided. The default value is correctly set. I have tested in both Chrome and Firefox.

I think the issue here was that using a binding like `$("Stock_Item_DP.Rows.0.Rated")` is potentially invalid, because you are assuming there is always a row - but this is not the case before the data has loaded. You would need to add optional chaining or check for nulls. The JS binding was therefore throwing an error and using the value `Error while executing JS` as the default value - causing the checkbox to be unchecked.

The skeleton loaders were the real culprit, as it was them that allowed these bindings to be evaluated before data had loaded. As we've reverted those loaders now this issue should be resolved.
